### ISSUE TITLE
Imagine My Surprise When Nullable Reference Types Became An Issue

### ISSUE THREAD

**Issue by 5U8cVs** [dotnet/docs#30348]
I pay a hefty fee to Microsoft to subscribe to Visual Studio and have done so for a long time.  I installed the latest creation the other day and compiled some of my old code in a new project.  Imagine my surprise when the compiler complained about nullable reference types.   Microsoft declined to send me a memo regarding substantial deprecations of syntax.  In fact, I *NEVER* hear from Microsoft.  The trade press sometimes mentions Microsoft when 97 security threats are discovered, many of them zero days.  I am a software developer and have been for many decades.  My old company took great care in documenting product changes.  Of course, my old company had only thousands of customers, not more than a billion.  My next employer was based in Paris.  They hired "writers" who were not technical but did speak English.  They conversed with developers and put words on paper.  Extracting product design from their documents, if you could get through them, was impossible.  Apparently, my industry, which is your industry, hates to write documentation.  Writing code is so much more enjoyable.  This page is good.  I'm glad I found it.  The error messages that Visual Studio issues when it discovers a reference type, that may or may not be null at runtime, could, but doesn't, refer to it.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a690704d-da32-dca7-ea3e-66b84fb1d00d
* Version Independent ID: 11f314de-dd3a-362e-b2bb-d0a2133cfc7b
* Content: [Nullable reference types](https://docs.microsoft.com/en-us/dotnet/csharp/nullable-references)
* Content Source: [docs/csharp/nullable-references.md](https://github.com/dotnet/docs/blob/main/docs/csharp/nullable-references.md)
* Product: **dotnet-csharp**
* Technology: **csharp-null-safety**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

---
[Associated WorkItem - 350754](https://dev.azure.com/msft-skilling/Content/_workitems/edit/350754)

**Comment by IEvangelist** [comment]
Hi @5U8cVs,

Thank you for posting this issue. I'm sorry that your experience isn't ideal, but I'm happy to help. When you say you opened some old code, can you tell me what version of .NET that code targets? How old? 

I think you're suggesting that Visual Studio should link to this page for these warnings you were seeing? Is that correct?

**Comment by 5U8cVs** [comment]
Hello David,

 

Thank you for your reply.  I appreciate it.  I started to develop a solution that depended on some old code.  So, I created a new solution under .NET 8.0 (that was the default at the time) and copied the code.  The compilation detected dozens of nullable reference errors.  So, I changed to a deprecated .NET (5.0 possibly).  And the problem went away.   I have been programming in C# for decades and the language suits my needs.  In most of my career I programmed the Univac 1100/2200 Series mainframes in assembly language which also suited my needs.  When I was forced off the mainframe I re-examined what Microsoft was offering and came away disappointed.  Visual Basic reeks of Dartmouth.  The C language is old hat.   The C++ language with Component Object Model has a very steep learning curve.  C# is well thought out and still undergoing development.   I especially like the helpful artificial intelligence features that speed up coding.

 

As for error messages coming out of Visual Studio, yes I am suggesting that error messages should link to documentation when possible.    Of course there is a code number already associated with every error message.  So one can Google the code number and acquire the link to the documentation.

 

My complaint is that my introduction to the problem of nullable reference types was sudden, unexpected, and it was unclear immediately how to cope with it.  In DEV/OPS there is supposed to be an organized collaboration between developers and the operational personnel who use what developers develop.  Microsoft publishes software and has claimed to be using DEV/OPS.  I am an end user AND a developer.  So sudden shifts in design emphasis should not occur.   The collaboration part isn’t happening for me.

 

Best regards,

 

Mark
